### PR TITLE
load spotify playlists from spotcast sensor

### PR DESCRIPTION
**Pull request by landonr**
for issue https://github.com/landonr/esphome-remote/issues/13
uses spotcast sensor https://github.com/fondberg/spotcast#use-the-sensor
use ArduinoJSON for decoding sensors with json
add TV to soundbar sources
fix soundbar not showing up in media players menu if it was playing tv 
move friendlyNameEntity to a new file and use it more

### CODE REVIEW

**Review by abroooo — COMMENTED** [review]
Don't know if you wanted me to review but I just do it 😄  Looks good to me so far. I just have a few suggestions/questions though.
Maybe it's worth mentioning that I'm still not that familiar with a lot of parts in the code yet especially the MediaPlayer part.

I think it would also be good to update the Readme/Getting Started because it requires a certain configuration on the home assistant side.

File: esphomeRemote/FriendlyNameEntity.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (landonr) [reply esphomeRemote/FriendlyNameEntity.h:1]: i saw this in another project and it seemed cleaner because it's only one line

**Comment by landonr** [comment]
@abroooo I appreciate the review! I moved all the text helpers to it's own class and made them static so we don't have to initialize the class. I also changed all the ifndef's to pragma once

File: esphomeRemote/esphomeRemotePlayer.h [hunk 2]
```diff
@@ -73,6 +73,56 @@ std::string filter(std::string str) {
   return output;
 }
 
+std::vector<MenuTitle> parseJsonArray(std::string state, std::string entityId) {
```
[the remark this answers is not in the document]
Comment (landonr) [reply esphomeRemote/esphomeRemotePlayer.h]: I created class TextHelpers and put other related text methods inside it

**Review by abroooo — APPROVED** [review]
I can't try it out as I haven't installed spotcast yet but it looks good 👍

**Pull request merged**